descartes_light/solvers/include/descartes_light/solvers/bgl/impl/event_visitors.hpp

```diff
@@ -32,6 +32,82 @@ struct early_terminator : public boost::base_visitor<early_terminator<FloatType>
   const long last_rung_idx_;
 };
 
+/**
+ * @brief Event visitor that adds all edges to each node discovered on a depth first search
+ * @details Throws the vertex descriptor that is the termination of the path once a vertex in the last rung of
+ * the graph is encountered
+ */
+template <typename FloatType>
+struct add_all_dfs : public boost::base_visitor<add_all_dfs<FloatType>>
```

Comment:
Rename to something more descriptive and accurate like `add_edges_dynamically`. This event visitor is not specific to DFS search